"listen on localhost" and "listen on all" doesn't work on Linux

--- a/smtpd/smtpd.conf
+++ b/smtpd/smtpd.conf
@@ -3,11 +3,11 @@
 # This is the smtpd server system-wide configuration file.
 # See smtpd.conf(5) for more information.
 
-table aliases file:/etc/mail/aliases
+table aliases file:/etc/smtpd/aliases
 
-# To accept external mail, replace with: listen on all
+# To accept external mail, add something like: listen on eth0
 #
-listen on localhost
+listen on lo
 
 action "local" mbox alias <aliases>
 action "relay" relay
